torch/_inductor/codegen/common.py

```diff
@@ -57,10 +57,15 @@ def _print_Pow(self, expr):
         if exp > 0:
             return "*".join([self.paren(base)] * exp)
         elif exp < 0:
-            return "1/" + self.paren("*".join([self.paren(base)] * abs(exp)))
+            return "1.0/" + self.paren("*".join([self.paren(base)] * abs(exp)))
         else:  # exp == 0
             return "1"
 
+    def _print_Rational(self, expr):
+        if expr.q == 1:
+            return f"{expr.p}"
```

Comment:
I guess this won't happen?

Reply:
It's required. While `Rational` should be able to optimize this, I get errors without this:

```
pytest -rfxs test/inductor/test_torchinductor_dynamic_shapes.py -k test_upsample_bilinear2d -v 

...

torch._dynamo.exc.BackendCompilerFailed: compile_fx_wrapper raised CompilationError: at 6:25:
def triton_(in_out_ptr0, in_ptr0, ks0, ks1, ks2, ks3, ks4, xnumel, XBLOCK : tl.constexpr):
    xoffset = tl.program_id(0) * XBLOCK
    xindex = xoffset + tl.arange(0, XBLOCK)[:]
    xmask = xindex < xnumel
    x1 = (xindex // ks0) % ks1
    x0 = (xindex // (1.0/1.0)) % ks0

...

error: invalid types 'float*[double]' for array subscript
  105 |                 auto tmp0 = out_ptr0[i1 + ((2.0/1.0)*i0*ks2)];
```

Reply:
This happens because we're not calling the `Rational` constructor directly. Exprs gets rewritten during printing:

```
[ins] In [7]: cexpr(1/(2*ks2 - 1))
Pow called 1/(2*ks2 - 1) Pow(Add(Mul(Integer(2), Symbol('ks2', integer=True, positive=True)), Integer(-1)), Integer(-1))
Rational called -1 Integer(-1)
Rational called 2 Integer(2)
Out[7]: '1.0/(((-1.0/1.0) + ((2.0/1.0)*ks2)))'
```

vs (with `== 1`):

```
[ins] In [5]: cexpr(1/(2*ks2 - 1))
Pow called 1/(2*ks2 - 1) Pow(Add(Mul(Integer(2), Symbol('ks2', integer=True, positive=True)), Integer(-1)), Integer(-1))
Rational called -1 Integer(-1)
Rational called 2 Integer(2)
Out[5]: '1.0/(((-1) + (2*ks2)))'
```

Reply:
`(xindex // (1.0/1.0))` doesn't look right, it should have been simplified in ModularIndexing, can you check why it wasn't? 
And again, seems like this change should be localized to CppPrinter

Reply:
@ngimel 

> (xindex // (1.0/1.0)) doesn't look right, it should have been simplified in ModularIndexing, can you check why it wasn't?

It's handled by `_print_ModularIndexing`, but it was printed like that due to me adding `print_Rational` (when I didn't have `if expr.q == 1`).

```py
import sympy
from torch._inductor.ir import ModularIndexing
from torch._inductor.codegen.triton import pexpr

print(pexpr(ModularIndexing(sympy.Symbol('xindex', integer=True, positive=True), sympy.Integer(1), sympy.Symbol('ks0', integer=True, positive=True))))
```

Output (with some debug prints added by me):
```
Calling _print_ModularIndexing (PythonPrinter) | ModularIndexing(xindex, 1, ks0) | ModularIndexing(Symbol('xindex', integer=True, positive=True), Integer(1), Symbol('ks0', integer=True, positive=True))
... printing x xindex <class 'sympy.core.symbol.Symbol'> Symbol('xindex', integer=True, positive=True)
... printing div 1 <class 'sympy.core.numbers.One'> Integer(1)
Calling _print_Rational | 1 | Integer(1)
... printing mod ks0 <class 'sympy.core.symbol.Symbol'> Symbol('ks0', integer=True, positive=True)
(xindex // (1.0/1.0)) % ks0
```

`sympy.One` inherits from `IntegerConstant`, which inherits from `Integer`, which inherits from `Rational`.

Reply:
> And again, seems like this change should be localized to CppPrinter

Okay, I'll move it there.